crates/common/logged_command/src/logged_command.rs

```diff
@@ -148,23 +148,34 @@ impl std::fmt::Display for LoggedCommand {
 }
 
 impl LoggedCommand {
-    pub fn new(program: impl AsRef<OsStr>) -> LoggedCommand {
+    pub fn new(program: impl AsRef<OsStr>) -> Result<LoggedCommand, shell_words::ParseError> {
         let command_line = match program.as_ref().to_str() {
             None => format!("{:?}", program.as_ref()),
             Some(cmd) => cmd.to_string(),
         };
 
-        let mut command = Command::new(program);
+        let mut args = shell_words::split(&command_line)?;
+
+        let mut command = match args.len() {
+            0 => return Err(shell_words::ParseError),
```

Comment:
The `Display` impl for `shell_words::ParseError` is `"missing closing quote"` which is very much not the case here. We should use our own error with a correct message.

Reply:
Also one more thing, need to add `time` feature to `logged_command/Cargo.toml`, because without it's impossible to run tests for the package.